core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServiceAction.kt

```diff
@@ -0,0 +1,125 @@
+package org.evomaster.core.problem.external.service
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.*
+import com.github.tomakehurst.wiremock.common.Metadata.metadata
+import org.evomaster.core.problem.external.service.param.ResponseParam
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.StructuralElement
+import org.evomaster.core.search.gene.Gene
+import org.evomaster.core.search.gene.GeneUtils
+
+/**
+ * Action to execute the external service related need
+ * to handle the external service calls.
+ *
+ * Typically, handle WireMock responses
+ */
+class ExternalServiceAction(
+
+    /**
+     * Received request to the respective WireMock instance
+     *
+     * TODO: Need to expand the properties further in future
+     *  depending on the need
+     */
+    val request: ExternalServiceRequest,
+
+    /**
+     * currently, we support response with json format
+     * then use ObjectGene now,
+     * might extend it later
+     */
+    val response: ResponseParam = ResponseParam(),
+
+    /**
+     * WireMock server which received the request
+     */
+    val wireMockServer: WireMockServer,
+    private val id: Long,
+
+    val representExistingRequest: Boolean = false
```

Comment:
add comments for this property

Reply:
I had this to track the same request appeared again, not has no purpose. Removing it.